proto/src/server.rs

```diff
@@ -198,29 +198,52 @@ impl<D: indradb::Datastore + Send + Sync + 'static> Server<D> {
     }
 }
 
+use tokio::runtime::Handle;
+use tokio::task::JoinError;
+
+fn map_tokio_result<T>(res: Result<T, JoinError>) -> Result<T, Status> {
+    res.map_err(|err| Status::internal(format!("{}", err)))
+}
+
 #[tonic::async_trait]
 impl<D: indradb::Datastore + Send + Sync + 'static> crate::indra_db_server::IndraDb for Server<D> {
     async fn ping(&self, _: Request<()>) -> Result<Response<()>, Status> {
         Ok(Response::new(()))
     }
 
     async fn sync(&self, _: Request<()>) -> Result<Response<()>, Status> {
-        map_indradb_result(self.datastore.sync())?;
+        let datastore = self.datastore.clone();
+
+        map_indradb_result(map_tokio_result(
+            Handle::current().spawn_blocking(move || datastore.sync()).await,
+        )?)?;
         Ok(Response::new(()))
     }
 
     async fn create_vertex(&self, request: Request<crate::Vertex>) -> Result<Response<crate::CreateResponse>, Status> {
+        let datastore = self.datastore.clone();
+
         let vertex = map_conversion_result(request.into_inner().try_into())?;
-        let res = map_indradb_result(self.datastore.create_vertex(&vertex))?;
+        let res = map_indradb_result(map_tokio_result(
+            Handle::current()
+                .spawn_blocking(move || datastore.create_vertex(&vertex))
```

Comment:
You should be able to just call `tokio::task::spawn_blocking`

Reply:
done. Thought it was better to be explicit, but I called it at the same callsite instead of holding the handle. Thanks - newer to tokio but done a lot of async work generally so happy for the style feedback.

Reply:
marked resolved.